src/main/webapp/populateData.js

```diff
@@ -385,16 +385,30 @@ function populatePlanMail() {
         averageReplyContainer.innerText = planMailResponse.averageReadingSpeed;
         const timeNeededContainer = document.querySelector('#time-needed');
         timeNeededContainer.innerText = planMailResponse.minutesToRead;
+        const messageEventContainer = document.querySelector('#message-event');
+        messageEventContainer.innerHTML = '';
         const intervalContainer = document.querySelector('#free-interval');
         intervalContainer.innerHTML = '';
-        for (const index in planMailResponse.potentialMeetingTimes) {
-          if (typeof index == 'string') {
-            const liElement = document.createElement('li');
-            liElement.innerText =
-              planMailResponse.potentialMeetingTimes[index].start +
-              ' ' +
-              planMailResponse.potentialMeetingTimes[index].end;
-            intervalContainer.appendChild(liElement);
+        if (planMailResponse.potentialEventTimes.length === 0) {
+          messageEventContainer.innerText = 'No new events needed';
+        } else {
+          messageEventContainer.innerText = 'Click to schedule';
+          for (const index in planMailResponse.potentialEventTimes) {
+            if (typeof index === 'string') {
+              const buttonElement = document.createElement('button');
+              buttonElement.innerText =
+                `${planMailResponse.potentialEventTimes[index].start} to \
+                ${planMailResponse.potentialEventTimes[index].end}`;
+              buttonElement.setAttribute('start',
+                  planMailResponse.potentialEventTimes[index].start);
+              buttonElement.setAttribute('end',
+                  planMailResponse.potentialEventTimes[index].end);
+              buttonElement.addEventListener('click', () => {
+                createEvent(buttonElement.getAttribute('start'),
+                    buttonElement.getAttribute('end'));
+              });
```

Comment:
You could simplify this by just capturing the start/end values in the event listener callback, but what you have is technically correct (and maybe useful in the future), so up to you.

Reply:
I think I tried this in the walkthrough projects. If I supply directly the values in the event listener, it will only capture the last values of the `const`. For safety purposes, I first linked the parameters as attributes of the button, that way the values are not impacted by the `for loop`.

Reply:
Oh yes, I see what the problem is. I think there is a way to work around it with temporary variables, but you can just leave it as is.